source/time_stepping/repeat_on_cutback.cc

```diff
@@ -0,0 +1,96 @@
+/*
+  Copyright (C) 2018 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/global.h>
+#include <aspect/time_stepping/repeat_on_cutback.h>
+
+namespace aspect
+{
+  namespace TimeStepping
+  {
+    template <int dim>
+    double
+    RepeatOnCutback<dim>::execute()
+    {
+      return std::numeric_limits<double>::max();
+    }
+
+
+
+    template <int dim>
+    Reaction
+    RepeatOnCutback<dim>::determine_reaction (const double time_step_size)
+    {
+      std::cout
+          << " old:" << this->get_old_timestep()
+          << " last: " << this->get_timestep()
+          << " now: " << time_step_size << std::endl;
+
+
+      if (time_step_size < this->get_timestep()*0.2)
+        return Reaction::repeat_step;
+      else
+        return Reaction::advance;
+    }
+
+
+
+    template <int dim>
+    void
+    RepeatOnCutback<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Time stepping");
+      {
+        prm.enter_subsection("Function");
```

Comment:
Rename `Function` to `Repeat on cutback`. The `0.2` factor could be made into a input parameter right?